ui/app/routes/jobs/job/variables.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// @ts-check
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+// eslint-disable-next-line no-unused-vars
+import JobModel from '../../../models/job';
+import { A } from '@ember/array';
+
+export default class JobsJobVariablesRoute extends Route {
+  @service can;
+  @service router;
+  @service store;
+
+  beforeModel() {
+    if (this.can.cannot('list variables')) {
+      this.router.transitionTo(`/jobs`);
+    }
+  }
+  async model() {
+    /** @type {JobModel} */
+    let job = this.modelFor('jobs.job');
+    let taskGroups = job.taskGroups;
+    let tasks = taskGroups.map((tg) => tg.tasks.toArray()).flat();
+
+    let jobVariable = await job.getPathLinkedVariable();
+    let groupVariables = await Promise.all(
+      taskGroups.map((tg) => tg.getPathLinkedVariable())
+    );
+    let taskVariables = await Promise.all(
+      tasks.map((task) => task.getPathLinkedVariable())
+    );
+
+    let allJobsVariable;
+    try {
+      let variables = await this.store.query('variable', {
+        path: 'nomad/jobs',
+      });
```

Comment:
You're waterfalling api calls by using `await` like this.

Since none of the requests depend on the responses from prior requests, something like this would let the browser handle how many connections it wishes to open in parallel:

```js
    // Capture the promises themselves as variables instead of the resolutions
    let jobVariablePromise = job.getPathLinkedVariable();
    let groupVariablesPromises =  taskGroups.map((tg) => tg.getPathLinkedVariable();
    let taskVariablesPromises = tasks.map((task) => task.getPathLinkedVariable()));

    let allJobsVariable;
    try {
      let variables = await this.store.query('variable', {
        path: 'nomad/jobs',
      });
    } /* ... */
    
    // And then let them all rip at once
    const variables = await Promise.all([
      allJobsVariablesPromise,
      jobVariablePromise,
      ...groupVariablesPromises,
      ...taskVariablesPromises,
    ]);
    /* ... */
```

And if @tgross is still checking in on this PR, this is kinda gnarly. It would be groovy if the API had a way to get the whole cascade of variables for a job in one API call.

Reply:
If we were only listing, we could use a [prefix](https://developer.hashicorp.com/nomad/api-docs/variables#prefix) on `nomad/jobs/:job_id` and get all the Variables in one API call. This is the usual "1+N queries" problem we have in the API where we only give stub objects on List. This does seems like it'd be a nice special case though.